pkg/upbin/upbin.go

```diff
@@ -0,0 +1,398 @@
+package upbin
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+	"time"
+	"unicode"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/storage/memory"
+	"github.com/manifoldco/promptui"
+	"gopkg.in/yaml.v2"
+
+	"github.com/aziontech/azion-cli/pkg/cmd/version"
+)
+
+type system string
+
+const (
+	azioncli string = "azioncli"
+
+	windows system = "windows"
+	darwin  system = "darwin"
+	linux   system = "linux"
+	unknown system = "unknown"
+
+	repoURL string = "https://github.com/aziontech/azion-cli.git"
+	url     string = "https://downloads.azion.com/%s/x86_64/azioncli"
+
+	LastActivity    string = "LAST_ACTIVITY"
+	packageAzioncli string = "aziontech/tap/azioncli"
+)
+
+// Variable factory for unit testing
+var (
+	notifyFunc           = notify
+	needToUpdateFunc     = needToUpdate
+	wantToUpdateFunc     = wantToUpdate
+	saveLastActivityFunc = saveLastActivity
+	prepareURLFunc       = prepareURL
+	managersPackagesFunc = managersPackages
+	whichFunc            = which
+	downloadFileFunc     = downloadFile
+	replaceFileFunc      = replaceFile
+
+	openConfigFunc = openConfig
+
+	packageManagerExistsFunc  = packageManagerExists
+	installPackageManagerFunc = installPackageManager
+)
+
+func UpdateBin() error {
+	notfy, err := notifyFunc()
+	if err != nil {
+		return err
+	}
+
+	if !notfy {
+		return nil
+	}
+
+	if !needToUpdateFunc() {
+		return nil
+	}
+
+	if !wantToUpdateFunc() {
+		return saveLastActivityFunc()
+	}
+
+	fileURL, err := prepareURLFunc()
+	if err != nil {
+		return err
+	}
+
+	install, err := managersPackagesFunc()
+	if err != nil {
+		return err
+	}
+
+	if install {
+		return nil
+	}
+
+	filePath, _ := whichFunc(azioncli)
+
+	err = downloadFileFunc(filePath, fileURL)
+	if err != nil {
+		return err
+	}
+
+	err = replaceFileFunc(filePath)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func GetCurrentVersion() int {
+	n, _ := Format(version.BinVersion)
+	return n
+}
+
+func GetLatestVersion() int {
```

Comment:
Can't we create a function that both this and the one in init command use?

Reply:
The idea is for init to use this function, I just didn't apply it 🪜